File: environment/env_v0_4/dirs.go
package env_v0_4

import (
	"fmt"
	"os"
	"path/filepath"
)

var rootDirName = ".zrok"

func SetRootDirName(name string) {
	rootDirName = name
}

func rootDir() (string, error) {
	home, err := os.UserHomeDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(home, rootDirName), nil
}

func metadataFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "metadata.json"), nil
}

func configFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "config.json"), nil
}

func environmentFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "environment.json"), nil
}

func identitiesDir() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "identities"), nil
}

func identityFile(name string) (string, error) {
	idd, err := identitiesDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(idd, fmt.Sprintf("%v.json", name)), nil
}

func agentSocket() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "agent.socket"), nil
}

func agentRegistry() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "agent-registry.json"), nil
}

func agentEnrollment() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "agent-enrollment.json"), nil
}

File: sdk/python/src/zrok/environment/dirs.py
from pathlib import Path
import os


def rootDir() -> str:
    home = str(Path.home())
    return os.path.join(home, ".zrok")


def metadataFile() -> str:
    zrd = rootDir()
    return os.path.join(zrd, "metadata.json")


def configFile() -> str:
    zrd = rootDir()
    return os.path.join(zrd, "config.json")


def environmentFile() -> str:
    zrd = rootDir()
    return os.path.join(zrd, "environment.json")


def identitiesDir() -> str:
    zrd = rootDir()
    return os.path.join(zrd, "identities")


def identityFile(name: str) -> str:
    idd = identitiesDir()
    return os.path.join(idd, name + ".json")
